tests/components/RangeInput.test.tsx

```diff
@@ -0,0 +1,260 @@
+    const minTextbox = screen.getAllByRole('textbox')[0];
+    userEvent.type(minTextbox, '10');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    userEvent.click(screen.getByText('Apply'));
+    expect(actions.setFilterOption).toHaveBeenCalledWith({
+      displayName: 'Over 10',
+      fieldId: '123',
+      matcher: '$between',
+      selected: true,
+      value: {
+        start: {
+          matcher: '$ge',
+          value: 10
+        },
+      }
+    });
+  });
+
+  it('renders correctly when clearing input and no state is set', async () => {
+    renderRangeInput();
+    const actions = spyOnActions();
+    const minTextbox = screen.getAllByRole('textbox')[0];
+    userEvent.type(minTextbox, '10');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(minTextbox).toHaveValue('');
+    expect(actions.setFilterOption).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe('Renders correctly for max input', () => {
+  beforeEach(() => {
+    mockAnswersHooks({ mockedState, mockedActions });
+  });
+
+  it('renders input value, clear, and apply option when inputing max value', async () => {
+    renderRangeInput();
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    expect(screen.getByText('Clear min and max')).toBeDefined();
+    expect(screen.getByText('Apply')).toBeDefined();
+  });
+
+  it('record proper values in state when applying max', async () => {
+    renderRangeInput();
+    const actions = spyOnActions();
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    userEvent.click(screen.getByText('Apply'));
+    expect(actions.setFilterOption).toHaveBeenCalledWith({
+      displayName: 'Up to 20',
+      fieldId: '123',
+      matcher: '$between',
+      selected: true,
+      value: {
+        end: {
+          matcher: '$le',
+          value: 20
+        },
+      }
+    });
+  });
+
+  it('renders correctly when clearing input and no state is set', async () => {
+    renderRangeInput();
+    const actions = spyOnActions();
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(maxTextbox).toHaveValue('');
+    expect(actions.setFilterOption).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe('Renders correctly for min and max inputs', () => {
+  beforeEach(() => {
+    mockAnswersHooks({ mockedState, mockedActions });
+  });
+
+  it('renders input values, clear, apply options and record proper values in state when applying min and max', async () => {
```

Comment:
I think it might be clearer to split this test up into one for checking rendering and one for checking what setFilterOption is called with

Reply:
Long names for things like variables, tests, and functions generally mean they're doing more than one thing and can be broken up. The general idea for breaking things up is called the "single responsibility principle", with the general idea being that small building blocks are easier to understand than big complicated things

Reply:
She had two separate tests in her original commit, but I made a comment to combine them to avoid duplicating the same setup since she had to type up all the values and confirm the necessary button(s) before checking actions.setFilterOption. The difference, beside checking for functions, might be not needing to check for `Clear min and max` label.

Reply:
I see, would it be possible to shorten the test name?